config/vufind/OAuth2Server.yaml

```diff
@@ -0,0 +1,110 @@
+  # (at least 32 characters)
+  encryptionKey: ""
+# Known clients configuration
+Clients:
+  example:
+    # Client name
+    name: Foo Client
+    # Redirect URI
+    redirectUri: "https://oauth-client/auth"
+    # Whether to use PKCE (see https://tools.ietf.org/html/rfc7636 for details). Must
+    # be enabled for non-confidential clients.
+    pkce: true
+    # Whether the client is confidential (as opposed to a public one such as an
+    # in-browser app)
+    isConfidential: false
+    # Hash of a client secret. You can use the following command to get a hash for a
+    # password: php -r 'echo password_hash("secret",  PASSWORD_DEFAULT) . PHP_EOL;'
+    secret: ""
+# Scope configuration. Keys are scope identifiers. Each identifier should include a
+# description (translation key) to be displayed to the user. The ils field should be
+# set to true if the information comes from a user profile in the library system.
+# This allows VuFind to display the library login or card selection form when
+# appropriate.
+Scopes:
+  openid: # openid is a built-in set required for OpenID
+    description: external_auth_scope_openid
+  id:
+    description: external_auth_scope_unique_id
+  name:
+    description: external_auth_scope_full_name
+  given_name:
+    description: external_auth_scope_given_name
+  family_name:
+    description: external_auth_scope_family_name
+  email:
+    description: external_auth_scope_email
+  birthdate:
+    description: external_auth_scope_birthdate
+    ils: true
+  locale:
+    description: external_auth_scope_locale
+  phone:
+    description: external_auth_scope_phone
+    ils: true
+  address:
+    description: external_auth_scope_address
+    ils: true
+# Claim set configuration. Claim sets map scopes to sets of user attributes.
+# Note that the OpenID Connect specification defines several standard claims
+# (see https://openid.net/specs/openid-connect-core-1_0.html#StandardClaims).
+# Note also that some or most of the information may only be available when the user
+# has an active library card.
+ClaimSets:
+  id:
+    - id
+  openid: # openid is a built-in set required for OpenID
+    - name
+    - email
+    - locale
+    - nonce # This is required for OpenID nonce check
+  name:
+    - name
+  given_name:
+    - given_name
+  family_name:
+    - family_name
+  email:
+    - email
+  birthdate:
+    - birthdate
+  locale:
+    - locale
+  phone:
+    - phone
+  address:
+    - address
+# Mappings from claim id to VuFind user or patron profile fields
+ClaimMappings:
+  id: id
+  name: full_name # special field created from firstname + ' ' + lastname
+  given_name: firstname
+  family_name: lastname
+  email: email
+  birthdate: birthdate
+  locale: last_language
+  phone: phone
+  address: address_json # special field including all address fields
+# Grants configuration
+Grants:
+  # Life time for an authorization code (should be kept short). Default is 1 minute.
+  authCodeLifeTime: PT1M
```

Comment:
I'm not familiar with the `PT...` syntax. Does this need to be explained in the comments?

Reply:
Comment expanded.